js/Source/Request/Request.PollForUpdate.js

```diff
@@ -63,10 +63,10 @@ Request.PollForUpdate = new Class({
     var data = this.request.response.json;
     // if the server provides an 'updated_at' timestamp,
     // store as an attribute
-    if (data.status == 'update'){
+    if (typeof data !== 'undefined' && data.status == 'update'){
```

Comment:
yeah, but you can express your check by just doing my line above. JS does type coercion for you (a pro and a con), so, for example:
```
!null == true
!false == true
!undefined == true
!"" == true
!0 == true
```
This coercion can get you into trouble of course, but in most cases - esp. where you're just checking that the value is present - it's a good shorthand and thus is a common style. It's also the reason why you should use `===` when you want something to be explicitly equal, especially if you're checking if it's false. For example:

```
undefined == null // true, though they are not the same thing
var x = {
  y: null
};
x.y == undefined //true
x.y === undefined //false - it IS defined
```

Anyway, even if you wanted to confirm that it is undefined here, you could do

```
if (data == undefined && ...
```

as `undefined == undefined` works (i.e. you don't need `typeof`)